packages/bsky/src/services/feed/views.ts

```diff
@@ -83,13 +84,18 @@ export class FeedViews {
 
   formatFeed(
```

Comment:
the 3rd party block check in here kinda seems like it should be in the rules fn instead of in the presentation fn 🤔

Reply:
altho i guess the problem there is that it technically stays in the skeleton but is "presented" differently (with `#notFoundPost`)

Reply:
I don't disagree at all, but it's tricky to make it happen without some more in-depth changes—same goes for the `noRules` on `getPostThread`.  As far as block state goes, luckily I think those are really the only two loose ends.  My primary goal here was to unify application and presentation of the viewer's block state.

Another similar situation is the presentation logic that happens during hydration of embeds.  Ideally all that presentation logic would be hoisted up too.

So you can think of this refactor as sort of going "one level deep", but there are some things in here that are deeply enough coupled to hydration that I chose not to dismantle.  Happy to hop back in though 😈

Reply:
Looking at this again, it might not bee too bad—I will give it a quick whirl!  Unraveling the embed presentation still looks like it may be for another day, though 🙃

Reply:
Dove into this, and sadly I think it makes sense to leave as-is for the time being.  It is actually straightforward to get rid of that one 3p block check for replies.  However, in `formatMaybePostView()` a similar check is made and in that case it applies to the presentation, possibly resulting in a `#blockedPost` for `post.reply.parent` rather than being omitted entirely.  Since these 3p block checks are so similar and appear nearby each other, I think it may be best to leave both of them colocated for now.

My takeaways:
 - it probably makes sense to blow this up when we do a deeper fix for handling 3p blocks (ref: https://github.com/bluesky-social/atproto/pull/1547#discussion_r1316434594) rather than this one-off.
 - may need to think through pipeline `rules()` more, to better account for situations where what we think of as rules end-up affecting the presentation beyond just omitting some data.